react-app/src/components/Room.js

```diff
@@ -1,19 +1,36 @@
-import React from 'react';
-import { UncontrolledCollapse, Button, CardBody, Card } from 'reactstrap';
+import React, { Component } from 'react';
+import { Collapse, Button, CardBody, Card } from 'reactstrap';
+import Timetable from './Timetable'
+class Room extends Component {
+  constructor(props) {
+    super(props);
+    this.toggle = this.toggle.bind(this);
+    this.state = { 
+      collapse: false,
+      roomName: props.roomName
+    };
+  }
```

Comment:
remove the `constructor` method. the latest React actually doesn't require any of this, especially the `this.toggle = this.toggle.bind(this)` is actually not necessary; @utkarsh867 actually told me that if you need to write this line of code, it actually means the function is "broken".

For `this.state`, refer to `InfoSection.js` and use this instead:
```
state = {
    helpCollapse: false,
    contactCollapse: false,
    aboutCollapse: false
  };
```

Reply:
In this case, how to pass props without a constructor?

I agree to the this.toggle = this.toggle.bind(this) removal part